Makefile

```diff
@@ -1,17 +1,38 @@
-TARGETS = rrdp_iostat \
-	 rrdp_squeezed \
-	 rrdp_xenpm
+DESTDIR ?=
+ETCDIR ?= /etc/xensource
+LIBEXECDIR ?= /opt/xensource/libexec
 
-.PHONY: build
-build: $(TARGETS)
+SETUP = ocaml setup.ml
 
-$(TARGETS):
-	omake $@
+build: setup.data
+	$(SETUP) -build $(BUILDFLAGS)
+
+doc: setup.data build
+	$(SETUP) -doc $(DOCFLAGS)
+
+test: setup.data build
+	$(SETUP) -test $(TESTFLAGS)
 
-.PHONY: clean
 clean:
-	omake clean
+	$(SETUP) -clean $(CLEANFLAGS)
+
+distclean:
+	$(SETUP) -distclean $(DISTCLEANFLAGS)
+
+setup.data:
+	$(SETUP) -configure $(CONFIGUREFLAGS)
+
+configure:
+	$(SETUP) -configure $(CONFIGUREFLAGS)
+
+.PHONY: build doc test clean distclean configure install
 
-.PHONY: install
-install: $(TARGETS)
-	omake install
+install: build
+	install -D -m 755 rrdp_iostat.native $(DESTDIR)$(LIBEXECDIR)/xcp-rrdd-plugins/xcp-rrdd-iostat
```

Comment:
They're set at the top of the Makefile if they're not already set.

In the build system, $DESTDIR is set in the specfile [here](https://github.com/xenserver/perf-tools/blob/master/SPECS/rrdd-plugins.spec#L35).